internal/mode/static/nginx/config/maps_test.go

```diff
@@ -189,3 +187,123 @@ func TestBuildAddHeaderMaps(t *testing.T) {
 }
+
+func TestExecuteStreamMaps(t *testing.T) {
+	g := NewWithT(t)
+	conf := dataplane.Configuration{
+		TLSPassthroughServers: []dataplane.Layer4VirtualServer{
+			{
+				Hostname:     "example.com",
+				Port:         8081,
+				UpstreamName: "backend1",
+			},
+			{
+				Hostname:     "example.com",
+				Port:         8080,
+				UpstreamName: "backend1",
+			},
+			{
+				Hostname:     "cafe.example.com",
+				Port:         8080,
+				UpstreamName: "backend2",
+			},
+		},
+		SSLServers: []dataplane.VirtualServer{
+			{
+				Hostname: "app.example.com",
+				Port:     8080,
+			},
+		},
+		StreamUpstreams: []dataplane.Upstream{
+			{
+				Name: "backend1",
+			},
+			{
+				Name: "backend2",
+			},
+		},
+	}
+
+	expSubStrings := map[string]int{
+		"example.com unix:/var/run/nginx/example.com8081.sock;":           1,
+		"example.com unix:/var/run/nginx/example.com8080.sock;":           1,
+		"cafe.example.com unix:/var/run/nginx/cafe.example.com8080.sock;": 1,
+		"app.example.com unix:/var/run/nginx/https8080.sock;":             1,
+		"hostnames": 2,
+	}
+
+	results := executeStreamMaps(conf)
+	g.Expect(results).To(HaveLen(1))
+	result := results[0]
+
+	g.Expect(result.dest).To(Equal(streamConfigFile))
+	for expSubStr, expCount := range expSubStrings {
+		g.Expect(strings.Count(string(result.data), expSubStr)).To(Equal(expCount))
+	}
+}
+
+func TestCreateStreamMaps(t *testing.T) {
+	g := NewWithT(t)
+	conf := dataplane.Configuration{
+		TLSPassthroughServers: []dataplane.Layer4VirtualServer{
+			{
+				Hostname:     "example.com",
+				Port:         8081,
+				UpstreamName: "backend1",
+			},
+			{
+				Hostname:     "example.com",
+				Port:         8080,
+				UpstreamName: "backend1",
+			},
+			{
+				Hostname:     "cafe.example.com",
+				Port:         8080,
+				UpstreamName: "backend2",
+			},
+		},
+		SSLServers: []dataplane.VirtualServer{
+			{
+				Hostname: "app.example.com",
+				Port:     8080,
+			},
+			{
+				Port:      8080,
+				IsDefault: true,
+			},
+		},
+		StreamUpstreams: []dataplane.Upstream{
+			{
+				Name: "backend1",
+			},
+			{
+				Name: "backend2",
+			},
+		},
+	}
+
+	maps := createStreamMaps(conf)
+
+	expectedMaps := []shared.Map{
+		{
+			Source:   "$ssl_preread_server_name",
+			Variable: getTLSPassthroughVarName(8081),
+			Parameters: []shared.MapParameter{
+				{Value: "example.com", Result: "unix:/var/run/nginx/example.com8081.sock"},
+			},
+			UseHostnames: true,
+		},
+		{
+			Source:   "$ssl_preread_server_name",
+			Variable: getTLSPassthroughVarName(8080),
+			Parameters: []shared.MapParameter{
+				{Value: "example.com", Result: "unix:/var/run/nginx/example.com8080.sock"},
+				{Value: "cafe.example.com", Result: "unix:/var/run/nginx/cafe.example.com8080.sock"},
+				{Value: "app.example.com", Result: "unix:/var/run/nginx/https8080.sock"},
+				{Value: "default", Result: "unix:/var/run/nginx/https8080.sock"},
+			},
+			UseHostnames: true,
+		},
+	}
+	g.Expect(maps).To(Equal(expectedMaps))
```

Comment:
Missing a couple test cases:
1. TLSPassthroughServers is empty -> should return nil
2. Upstream does not exist -> should set socket to ""